Framework/Crystal/test/PredictSatellitePeaksTest.h

```diff
@@ -159,13 +160,103 @@ class PredictSatellitePeaksTest : public CxxTest::TestSuite {
     TS_ASSERT(alg.isExecuted());
 
     alg.setPropertyValue("SatellitePeaks", "SatellitePeaks");
-    PeaksWorkspace_sptr SatellitePeaks2 = alg.getProperty("SatellitePeaks");
+
+    IPeaksWorkspace_sptr SatelliteIPeaks2 = alg.getProperty( "SatellitePeaks" );
+    PeaksWorkspace_sptr SatellitePeaks2 = 
+    std::dynamic_pointer_cast<PeaksWorkspace>(SatelliteIPeaks2);
+
     TS_ASSERT(SatellitePeaks2);
+    TS_ASSERT_EQUALS( SatellitePeaks2->getNumberPeaks(), 4 );
 
     /* pull out the satellite peaks */
     std::vector<Peak> const &satellite_peaks = SatellitePeaks2->getPeaks();
 
-    TS_ASSERT_EQUALS(satellite_peaks.size(), 4);
+    //TS_ASSERT_EQUALS(satellite_peaks.size(), 4);
+
+    /* check goniometer matrices */
+
+    TS_ASSERT_EQUALS(satellite_peaks[0].getGoniometerMatrix(), Mantid::Kernel::Matrix<double>(3, 3, true));
+
+    TS_ASSERT_EQUALS(satellite_peaks[1].getGoniometerMatrix(), Mantid::Kernel::Matrix<double>(3, 3, true));
+
+    TS_ASSERT_EQUALS(satellite_peaks[2].getGoniometerMatrix(),
+                     Mantid::Kernel::Matrix<double>({-1, 0, 0, 0, 1, 0, 0, 0, -1}));
+
+    TS_ASSERT_EQUALS(satellite_peaks[3].getGoniometerMatrix(),
+                     Mantid::Kernel::Matrix<double>({-1, 0, 0, 0, 1, 0, 0, 0, -1}));
+
+    /* check peak HKL coordinates */
+    TS_ASSERT_EQUALS(satellite_peaks[0].getHKL(), Mantid::Kernel::V3D(1.2, 1, 0));
+    TS_ASSERT_EQUALS(satellite_peaks[1].getHKL(), Mantid::Kernel::V3D(0.8, 1, 0));
+    TS_ASSERT_EQUALS(satellite_peaks[2].getHKL(), Mantid::Kernel::V3D(-1.2, -1, 0));
+    TS_ASSERT_EQUALS(satellite_peaks[3].getHKL(), Mantid::Kernel::V3D(-0.8, -1, 0));
+
+    /* check run numbers */
+    TS_ASSERT_EQUALS(satellite_peaks[0].getRunNumber(), 0);
+    TS_ASSERT_EQUALS(satellite_peaks[1].getRunNumber(), 0);
+    TS_ASSERT_EQUALS(satellite_peaks[2].getRunNumber(), 1);
+    TS_ASSERT_EQUALS(satellite_peaks[3].getRunNumber(), 1);
+
+    AnalysisDataService::Instance().remove("peaks");
+  }
+  
+  /* Same test as above, but with LeanPeaksWorkspace instead of PeaksWorkspace */
+  void test_exec_multiple_goniometers_lean() {
+    /* create a TOPAZ instrument */
+    FrameworkManager::Instance().exec("LoadEmptyInstrument", 4, "InstrumentName", "TOPAZ", "OutputWorkspace", "inst");
+
+    /* Add it to a Peaks workspace */
+    FrameworkManager::Instance().exec("CreatePeaksWorkspace", 6, "InstrumentWorkspace", "inst", "NumberOfPeaks", "0",
+                                      "OutputWorkspace", "peaks", 
+                                      "OutputType", "LeanElasticPeak");
+
+    /* retrieve a handle to the peaks workspace created in the FrameworkManager
+     */
+    LeanElasticPeaksWorkspace_sptr peaks_workspace;
+    TS_ASSERT_THROWS_NOTHING(
+        peaks_workspace = std::dynamic_pointer_cast<LeanElasticPeaksWorkspace>(AnalysisDataService::Instance().retrieve("peaks")));
+
```

Comment:
add a `TS_ASSERT(peaks_workspace)` here and you'll see the problem has something to do with this cast.